sys-utils/mountpoint.c

```diff
@@ -182,18 +192,18 @@ int main(int argc, char **argv)
 
 	ctl.path = argv[optind];
 
-	if (stat(ctl.path, &ctl.st)) {
+	if (lstat(ctl.path, &ctl.st)) {
 		if (!ctl.quiet)
 			err(EXIT_FAILURE, "%s", ctl.path);
 		return EXIT_FAILURE;
 	}
-	if (ctl.dev_devno)
-		return print_devno(&ctl) ? EXIT_FAILURE : EXIT_SUCCESS;
-	if (dir_to_device(&ctl)) {
+	if ((ctl.nofollow && S_ISLNK(ctl.st.st_mode)) || dir_to_device(&ctl)) {
 		if (!ctl.quiet)
 			printf(_("%s is not a mountpoint\n"), ctl.path);
 		return EXIT_FAILURE;
 	}
+	if (ctl.dev_devno)
+		return print_devno(&ctl) ? EXIT_FAILURE : EXIT_SUCCESS;
```

Comment:
I don't think you can move if (ctl.dev_devno), it changes -x semantic for non-nofollow...

Reply:
Right. --nofollow and -x, --devno need to be mutually exclusive. I'll send updated version before Monday.